### ISSUE TITLE
Cluseter-Scope directory structure has inconsistencies 

### ISSUE THREAD

**Issue by HumairAK** [operate-first/apps#610]
This particularly affects the `base` directory

Initially we set out to sort things according to this schema: 

`cluster-scope/base/<resource_plural_name>/<resource_metadata_name>/<resource_singular_name>.yaml` 

However this doesn't take into account `api_groups` which can introduce conflicts, some examples include kinds named `Config` or `Projects` of which there are multiple, belonging to different api groups. So we may have: 

- A resource of kind `Project` named `example-project` with an apiversion: `project.openshift.io/v1`
- A resource of kind `Project` named `example-project` with an apiversion: `config.openshift.io/v1`

Both of these can exist in the cluster simultaneously but not in our git repo if we are to respect our current schema, so we should update our structuring to factor in apiversions.

**Comment by HumairAK** [comment]
@larsks @tumido

**Comment by larsks** [comment]
Are you thinking of a directory layout that would be something like...

```
cluster-scope/
  base/
    project.openshift.io/
      v1/
        projects/
          example-project/
            kustomization.yaml
            project.yaml
    config.openshift.io/
      v1/
        projects/
          example-project/
            kustomization.yaml
            project.yaml
```

?

**Comment by HumairAK** [comment]
@larsks -- this is what I was thinking, but I think @tumido was against this due to the additional layers of nesting. The other suggestion would be to include the api/version in the name somehow, maybe something like: 

`cluster-scope/base/<API_VERSION_resource_plural_name>/<resource_metadata_name>/<resource_singular_name>.yaml`

**Comment by tumido** [comment]
Hm.. to me it would be the best if we're not inventing yet another custom structure and rather just try fitting the scheme to what we know or to what is being widely adopted elsewhere. And then just fit that to our `kustomize` paradigm of `base`s, `components`, and `overlays`.

I took a look into mustgather for example. They have some duplication of manifests as well and sometimes they insist on putting multiple resources into the same file, but in general I think the structure makes somewhat sense:

```
.
├── cluster-scoped-resources
│   ├── admissionregistration.k8s.io
│   │   ├── mutatingwebhookconfigurations
│   │   └── validatingwebhookconfigurations
│   ├── apiextensions.k8s.io
│   │   └── customresourcedefinitions
│   │       ├── alertmanagerconfigs.monitoring.coreos.com.yaml
...
├── namespaces
│   ├── default
│   │   ├── app.k8s.io
│   │   ├── apps
│   │   ├── apps.openshift.io
│   │   ├── autoscaling
...
```

They are probably trying to avoid folder/file name conflicts as hard as we do. I assume that it led them to adopting this structure:

`cluster-scoped-resources/<API_GROUP>/<KIND_PLURAL>/<NAME>.yaml`

In our case this would look like:

`cluster-scoped-resources/base/<API_GROUP>/<KIND_PLURAL>/<NAME>.yaml`

And when included in an overlay it can look like this:

```yaml
resources:
  - ../../../base/apiextensions.k8s.io/customresourcedefinitions/kfdefs.kfdef.apps.kubeflow.org
```
or:

```yaml
resources:
  - ../../../base/apiextensions.k8s.io/customresourcedefinitions/kfdefs.kfdef.apps.kubeflow.org.yaml
```
depending if we want to keep/drop the blank `kustomization` files and rather sacrifice on overlay `kustomization` resources uniformity (some entries will have `.yaml` suffix, some won't)...

I think the resource pointer in overlay will get quite long and possibly error prone, however if we shoot for automated/scripted solution of adding these, I think this aspect is not that much of an issue.

**Comment by HumairAK** [comment]
hrmm I like the idea of dropping the kustomization files, we already have consistencies in the overlay kustomizations any way: https://github.com/operate-first/apps/blob/master/cluster-scope/overlays/moc/zero/kustomization.yaml#L133

**Comment by HumairAK** [comment]
so to round up this discussion, are we good with: 

`cluster-scoped-resources/<API_GROUP>/<KIND_PLURAL>/<NAME>.yaml` ? then pulling the files in (with `.yaml` needing to be specified in overlays?) ?

**Comment by larsks** [comment]
> cluster-scoped-resources/<API_GROUP>/<KIND_PLURAL>/<NAME>.yaml ? then pulling the files in (with .yaml needing to be specified in overlays?) ?

Meaning we'd end up with something like:

```
resources:
  - ../../../base/rbac.authorization.k8s.io/clusterroles/argocd-manager.yaml
  - ../../../base/apiextensions.k8s.io/crds/cluster.ray.io.yaml
```

Etc?

**Comment by tumido** [comment]
yeah, let's simplify it as much as possible. :slightly_smiling_face: let's do that. 

@larsks yes, I least that's what I imagine now. To be precise, rather:
```
resources:
  - ../../../base/rbac.authorization.k8s.io/clusterroles/argocd-manager.yaml
  - ../../../base/apiextensions.k8s.io/customresourcedefinitions/cluster.ray.io.yaml
```

**Comment by larsks** [comment]
Where do core resource types go?  E.g, for...

```
apiVersion: v1
kind: Namespace
metadata:
  name: example
```

Do this go in `base/namespaces/example.yaml`? Or `base/core/namespaces/example.yaml`? Or...?

**Comment by tumido** [comment]
Interesting question... I'd say directly to the base. Like if api group is empty than there's no need for a special folder. WDYT?

**Comment by HumairAK** [comment]
yeah, we can keep it in `/base/`

**Comment by HumairAK** [comment]
yep sure

**Comment by tumido** [comment]
#622 merged, closing as resolved :slightly_smiling_face:
